csrc/device_lower/pass/scalar_hoist.cpp

```diff
@@ -339,9 +339,24 @@ std::vector<Val*> getAssumptions(const std::vector<kir::ForLoop*>& loops) {
     if (loop->isTrivial()) {
       continue;
     }
-    assumptions.push_back(
-        IrBuilder::ltExpr(loop->index(), loop->simplifiedStop()));
-    assumptions.push_back(IrBuilder::geExpr(loop->index(), loop->start()));
+    Val* start = loop->start();
+    assumptions.push_back(IrBuilder::geExpr(loop->index(), start));
+    Val* stop = loop->simplifiedStop();
+    if (stop->sameAs(start)) {
+      // If stop = start, then this loop will not be computed, so it's not
+      // important to simplify its index. However, it is important that we avoid
+      // contradicting assumptions, so we omit the index < stop assumption in
+      // these cases.
+      TORCH_WARN_ONCE(
+          "Encountered loop with no iterations. Stop value ",
+          stop->toInlineString(),
+          " is same as start value ",
+          start->toInlineString(),
+          ". This could indicate a suboptimal schedule such as double-buffering a ",
+          "loop that has only a single iteration.");
```

Comment:
Warning looks like 
```
[W scalar_hoist.cpp:350] Warning: Encountered loop with no iterations. Stop value 0 is same as start value 0. This could indicate a suboptimal schedule such as double-buffering a loop that has only a single iteration. (function operator())
```